backend/agents/persona_agent.py

```
@@ -120,40 +122,286 @@ def _create_persona_tools(self) -> List[BaseTool]:
+                        
+                        if history_parts:
+                            # Log the first few history parts to see what's being retrieved
+                            print(f"[DEBUG] First few history parts: {history_parts[:3]}")
+                            return f"Complete conversation history for this scene:\n" + "\n".join(history_parts)
+                        else:
+                            return "No conversation history available for this scene"
+                    else:
+                        print(f"[DEBUG] get_conversation_history - No docs found")
+                        return "No recent conversation history available"
+                else:
+                    raise ValueError("PGVector not available - vectorstore is required")
```

Ruff (0.14.0)

Abstract `raise` to an inner function

(TRY301)